### DISCUSSION TITLE
Material for MkDocs `v9.1.12` appears to break the abbreviation tooltips on hover

### DISCUSSION THREAD

**Discussion by HonkingGoose** [squidfunk/mkdocs-material#5529] [General]
First off, thanks again for making such cool software with you and your contributors! ❤️ 

I found a possible regression in Material for MkDocs `v9.1.12`. Summary:

- It looks like Material for MkDocs `v9.1.12` is the "bad commit"
- I marked `v9.1.11` as good with `git bisect good`
- Upgrading to `v9.1.13` makes the abbreviation tooltips work again

On the Renovate docs, the abbreviation tooltips [^docs] stopped working. I ran `git bisect` [^scm-docs] to find the commit that causes the problem. According to `git bisect` the problem is with the commit that updated us to `mkdocs-material v9.1.12`: [^quote]

> Here's the `git bisect log` output:
>
> ```shell
> # git bisect log
> git bisect start
> # status: waiting for both good and bad commits
> # bad: [c95f937c95d118e7b6de801fd297b672f2871923] chore(docs): update deps/renovate digest to b5d74a8
> git bisect bad c95f937c95d118e7b6de801fd297b672f2871923
> # status: waiting for good commit(s), bad commit known
> # good: [6978ff0bf3a49df62b583185f164994819bbe2d3] chore: show tooltip when hovering over abbreviation (#207)
> git bisect good 6978ff0bf3a49df62b583185f164994819bbe2d3
> # good: [9c7f84cca1d807ad4a2e4e7cbdc286a0e6d8515c] chore(docs): update deps/renovate digest to 6beeba6
> git bisect good 9c7f84cca1d807ad4a2e4e7cbdc286a0e6d8515c
> # good: [41d74eb7d98be911a99726fcadefbcef20559e4e] chore(docs): update deps/renovate digest to 56c161a
> git bisect good 41d74eb7d98be911a99726fcadefbcef20559e4e
> # good: [aea7a13eba8a2f7111167c5c66c717515f624901] chore(docs): update deps/renovate digest to c5a5c89
> git bisect good aea7a13eba8a2f7111167c5c66c717515f624901
> # good: [cae583a258f4226ea64d1deb52a590eab608fcec] chore(docs): update deps/renovate digest to b27ec63
> git bisect good cae583a258f4226ea64d1deb52a590eab608fcec
> # good: [522c36d212b02cfe1559d97823fc2d37632adad0] chore(docs): update deps/renovate digest to 1cd405b
> git bisect good 522c36d212b02cfe1559d97823fc2d37632adad0
> # good: [fa7e2c22f72940f69a57beac0f853c846407cbe8] chore(deps): update dependency mkdocs-material to v9.1.11
> git bisect good fa7e2c22f72940f69a57beac0f853c846407cbe8
> # good: [e63cec6587edd1f7b4605cb47e387a1dcdd91bfe] chore(docs): update deps/renovate digest to e74e480
> git bisect good e63cec6587edd1f7b4605cb47e387a1dcdd91bfe
> # bad: [b812b9c7acfe63da462c3242e4d248b16972a75b] chore(docs): update deps/renovate digest to f2977e4
> git bisect bad b812b9c7acfe63da462c3242e4d248b16972a75b
> # bad: [411f115201ef401673050a6815db929f3f0854bd] chore(deps): update dependency mkdocs-material to v9.1.12
> git bisect bad 411f115201ef401673050a6815db929f3f0854bd
> # good: [6c64ab9a36d4856e48837bf0779eaadfaeca8cbc] chore(docs): update deps/renovate digest to 85fbad1
> git bisect good 6c64ab9a36d4856e48837bf0779eaadfaeca8cbc
> # first bad commit: [411f115201ef401673050a6815db929f3f0854bd] chore(deps): update dependency mkdocs-material to v9.1.12
> ```

I know you usually want a minimal reproduction, which is a bit tricky, because we're patching in a lot of docs from other places in our build process. So giving you my `git bisect` results is the best that I can do, unfortunately. I still wanted to let you know that something broke.

[^quote]: https://github.com/renovatebot/renovatebot.github.io/issues/287
[^docs]: https://squidfunk.github.io/mkdocs-material/reference/tooltips/#adding-a-glossary
[^scm-docs]: https://git-scm.com/docs/git-bisect

**Comment by squidfunk** [comment]
Thanks for noting! That is extremely weird, because there were no changes whatsoever to the CSS or JS between version 9.1.12 and 9.1.13 ([check the diff](https://github.com/squidfunk/mkdocs-material/compare/9.1.12...9.1.13)). However, since you report that it's fixed in 9.1.13, I'm not sure we should invest time and investigate.

**Reply by squidfunk** [reply]
... however, I'm also not sure what you mean by "stopped working", because this could mean dozens of things. No `abbr` tags are generated? They don't appear? MkDocs crashes?

**Reply by HonkingGoose** [reply]
Thank you for the quick response! Your package is good. The problem appears to be with `pymdown-extensions` `10.0.0`.

### `v9.1.12` broke for Renovate but not for Git Gosling

`v9.1.12` broke the abbreviations for the Renovate docs site. But my simple Git Gosling site has _working_ abbreviations on that _same_ `v9.1.12` version.

### What's the difference between the repo updates?

The Renovate docs repository updated `mkdocs-material` **and** `pymdown-extensions` `10.0` together. But my simple docs **only** updated `mkdocs-material` to `9.1.12`.

### Why `mkdocs-material v9.1.13` appears to fix the problem

The Renovate update to `mkdocs-material v9.1.13` also pulled in a lockfile update for `pymdown-extensions` to `10.0.1`, which has a regression fix: "**FIX:** Regression related to snippets nested deeply under specified base path." [^fix]

### Timeline

1. [Git Gosling only updates `mkdocs-material` to `9.1.12`](https://github.com/HonkingGoose/git-gosling/commit/1bd496abf6a24ae3778030313ae3139c99c456e3) but.... [Renovate updates both `mkdocs-material` and `pymdown-extensions`](https://github.com/renovatebot/renovatebot.github.io/commit/411f115201ef401673050a6815db929f3f0854bd?short_path=a86c67a#diff-a86c67a0a29ed0e95909b9b7c420140f302d17399ee6dcce4e1a51a14d27fd51)
2. [Renovate updates `mkdocs-material` and pulls fix for `pymdown-extensions` at once](https://github.com/renovatebot/renovatebot.github.io/commit/6d2ac8cbf1a9b91ef93493d5701cd4b551f0934d?short_path=a86c67a#diff-a86c67a0a29ed0e95909b9b7c420140f302d17399ee6dcce4e1a51a14d27fd51)

[^fix]: https://github.com/facelessuser/pymdown-extensions/releases/tag/10.0.1

**Reply by HonkingGoose** [reply]
> ... however, I'm also not sure what you mean by "stopped working", because this could mean dozens of things. No abbr tags are generated? They don't appear? MkDocs crashes?

Steps to reproduce:

1. Open https://github.com/renovatebot/renovatebot.github.io/tree/411f115201ef401673050a6815db929f3f0854bd which is the commit that "broke things"
2. Click on the green <> Code button to open a GitHub Codespace
3. Wait for the container build to finish, then run `make && make serve` in the bash shell
4. Open the preview site, stay on the homepage, scroll to the _Supported Platforms_ list, and look at the AWS CodeCommit line, see that AWS is **not** underlined and does **not** show a tooltip when you hover over it with your mouse/pointer device
5. Open production site (which has the latest versions of `mkdocs-material` and `pymdown-extensions`) and see tooltip when you hover over the AWS abbreviation

**Reply by facelessuser** [reply]
Please update to pymdown-extensions 10.0.1. There was a regression in 10.0. If using snippets, it is expected the snippet will be a child of one of the base paths defined in `base_path`. It is recommended to to use 10.0.1, make sure you have reasonable base paths in `base_path` and ensure the snippets used in Markdown content are relative to one of those configured base paths and that they are children of said base path.

**Reply by HonkingGoose** [reply]
> Please update to pymdown-extensions 10.0.1

Hi @facelessuser 👋 Thank you for commenting! ❤️ The Renovate docs site now uses the latest version of both Material for MkDocs and your extension, and the abbreviations are working again.

I also want to apologize for bothering both you and @squidfunk with this.

**Reply by facelessuser** [reply]
Yep, sorry for the confusion.
